Server/Plugins/APIDump/APIDesc.lua

```diff
@@ -11689,6 +11797,40 @@ end
 					},
 					Notes = "Returns true if the specified player is queued to be transferred to a World.",
 				},
+				RegisterForgeMod =
+				{
+					Params =
+					{
+						{
+							Name = "ModName",
+							Type = "string",
+						},
+						{
+							Name = "ModVersion",
+							Type = "string",
+						},
+					},
+					Notes = "Add a Forge mod name/version to the server ping list.",
+				},
+				RegisterForgeModForProtocol =
+				{
+					Params =
+					{
+						{
+							Name = "ModName",
+							Type = "string",
+						},
+						{
+							Name = "ModVersion",
+							Type = "string",
+						},
+						{
+							Name = "ProtocolVersionNumber",
+							Type = "number",
+						},
+					},
+					Notes = "Add a Forge mod name/version to the server ping list for all protocol versions.",
```

Comment:
What if the Cuberite plugin was required to support all protocol versions supported by the server? I don't play with a lot of Forge mods, but the few I've seen all require the client to have the mod in order to join the server, the mod isn't optional (Imagine if the mod adds a block type, every client connecting would have to have the plugin, otherwise they wouldn't be able to see the block).

So then if the plugin only worked with some protocol versions, the server administrator could configure the server to only allow supported protocol versions (this is a thing that can happen, right? I thought I saw a PR about this somewhere).

Then we don't have to keep track of which forge mods support which protocols. There's just a single list of forge mods, which have to support all the protocols that the server supports.

I don't know if that's too limiting in terms of Forge mods, though. I don't know of any examples that benefit from per-protocolversion mods, but then I haven't seen many examples.